packages/connectivity/src/http-agent/http-agent.ts

```diff
@@ -59,58 +62,97 @@ function createProxyAgent(
-        passphrase: destination.keyStorePassword
-      };
+/*
+ Adds trust store certificate to the list of trusted CA so that https connection can also work with self signed certificates.
+ */
+function trustStoreOptions(
+  options: Record<string, any>,
+  destination: Destination
+): Record<string, any> {
+  if (destination.trustStoreCertificate) {
+    const filetype = destination.trustStoreCertificate.name.split('.')[1];
+    if (filetype.toLowerCase() !== 'pem') {
+      logger.warn(
+        `The provided truststore ${destination.trustStoreCertificate.name} is not in 'pem' format which is currently the only supported format. Trustore is ignored.`
+      );
+      return options;
     }
-    return options;
-  };
+    const decodedCertificate = new Buffer(
+      destination.trustStoreCertificate.content,
+      'base64'
+    ).toString('utf8');
+    return {
+      ...options,
+      ca: [decodedCertificate]
+    };
+  }
+  return options;
+}
+
 /**
  * @internal
- * The http agents (proxy and default) use node tls for the certificate handling. This method creates the options with the pfx and passphrase.
+ * The http agents (proxy and default) use node tls for trust handling. This method creates the options with the 'ca' or 'rejectUnauthorized' option.
  * https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
  * @param destination - Destination object
- * @returns Options, which can be used later by tls.createSecureContext() e.g. pfx and passphrase or an empty object, if the protocol is not 'https:' or no client information are in the definition.
+ * @returns Options, which can be used later the http client.
  */
-function getCertificateOption(destination: Destination): Record<string, any> {
+function getTrustStoreOptions(destination: Destination): Record<string, any> {
   // http case: no certificate needed
   if (getProtocolOrDefault(destination) === Protocol.HTTP) {
     if (destination.isTrustingAllCertificates) {
       logger.warn('"isTrustingAllCertificates" is not available for HTTP.');
     }
-
+    if (destination.trustStoreCertificate) {
+      logger.warn('"trustStore" is not available for HTTP.');
+    }
     return {};
   }
-  // https case: get certificate options
+
+  // https case
   if (destination.isTrustingAllCertificates) {
     logger.warn(
       '"isTrustingAllCertificates" property in the provided destination is set to "true". This is highly discouraged in production.'
     );
+    return { rejectUnauthorized: !destination.isTrustingAllCertificates };
   }
 
-  const options = trustAllOptions(destination)({});
-  return certificateOptions(destination)(options);
+  if (destination.trustStoreCertificate) {
+    const decoded = new Buffer(
+      destination.trustStoreCertificate.content,
+      'base64'
+    ).toString('utf8');
+    return {
+      rejectUnauthorized: true,
+      ca: [decoded]
+    };
+  }
+  return { rejectUnauthorized: true };
```

Comment:
I like how you adjusted this function 👍 
Just a thought: Should there be some kind of warning if `isTrustingAllCertificates` is set to true + `trustStoreCertificate` is set?